core/src/main/java/org/elasticsearch/search/suggest/completion/CompletionSuggestionBuilder.java

```diff
@@ -117,35 +167,38 @@ public CompletionSuggestionBuilder regex(String regex, RegexOptions regexOptions
      * Note: Only doc values enabled fields are supported
      */
     public CompletionSuggestionBuilder payload(List<String> fields) {
-        this.payloadFields.addAll(fields);
+        this.payloadFields = fields;
         return this;
     }
 
     /**
-     * Sets query contexts for a category context
-     * @param name of the category context to execute on
-     * @param queryContexts a list of {@link CategoryQueryContext}
+     * Sets query contexts for completion
+     * @param queryContexts named query contexts
+     *                      see {@link org.elasticsearch.search.suggest.completion.context.CategoryQueryContext}
+     *                      and {@link org.elasticsearch.search.suggest.completion.context.GeoQueryContext}
      */
-    public CompletionSuggestionBuilder categoryContexts(String name, CategoryQueryContext... queryContexts) {
```

Comment:
Was this setter just introduced with the new completion suggester on master or is this a change in the java api that we should track in the migrate.asciidoc?